### ISSUE TITLE
Warning suppression can't be overridden with base settings

### ISSUE THREAD

**Issue by robmaceachern** [tuist/tuist#6707]
### What happened?

I was attempting to un-suppress warnings for the frameworks defined in my Package.swift with `GCC_WARN_INHIBIT_ALL_WARNINGS` and `SWIFT_SUPPRESS_WARNINGS` in the `PackageSettings.baseSettings` but the values do not seem to be applied to the generated targets.

Other values, such as `APPLICATION_EXTENSION_API_ONLY` or `TESTING123` _are_ successfully applied to the generated targets when specified in `baseSettings`.

### How do we reproduce it?

1. Configure a Package.swift with the following base settings:

```
    baseSettings: .settings(
        base: [
            "APPLICATION_EXTENSION_API_ONLY": "$(inherited)", // Does get applied
            "GCC_WARN_INHIBIT_ALL_WARNINGS": "NO", // Does not get applied
            "SWIFT_SUPPRESS_WARNINGS": "NO", // Does not get applied
            "TESTING123": "YES", // Does get applied
        ]
    ),
```

2. Generate the project and open
3. Examine the build settings for `GCC_WARN_INHIBIT_ALL_WARNINGS` and `SWIFT_SUPPRESS_WARNINGS` in Xcode

Expected: They are both NO

Actual: They are both YES.

### Error log

No error log

### macOS version

14.6.1

### Tuist version

4.23.0

### Xcode version

15.4

**Comment by ajkolean** [comment]
Not sure if I am right but this mu current understanding. **Tuist** uses **SPM** for package resolution underneath, and **SPM** handles certain settings. However, **SPM drops or ignores Xcode-specific settings** like **`GCC_WARN_INHIBIT_ALL_WARNINGS`** and **`SWIFT_SUPPRESS_WARNINGS`** because these settings are tied to **Xcode’s build system and toolchain**, which **SPM** doesn’t manage.

Since **SPM** is focused on **Swift** and **cross-platform builds**, it doesn’t propagate these Xcode-specific flags, meaning they are lost before **Tuist** can apply them. **Tuist** itself is not responsible for this behavior.

If it's necessary you could move these settings to **`targetSettings`** in your **`Package.swift`**. **Tuist** injects target-level settings into the generated Xcode project’s **Build Settings**, ensuring that these flags are applied properly.

```swift
targetSettings: [
     "EACH_SPM_TARGET": ["SWIFT_SUPPRESS_WARNINGS": "NO", "GCC_WARN_INHIBIT_ALL_WARNINGS": "NO"], 
]
```

### Why This Works:
- **`targetSettings`** apply directly to each target in the Xcode project, allowing **Tuist** to modify the **Build Settings** for that target. This ensures that Xcode-specific flags like **`GCC_WARN_INHIBIT_ALL_WARNINGS`** and **`SWIFT_SUPPRESS_WARNINGS`** are applied correctly.
- **`baseSettings`** rely on **SPM** to propagate settings, but since **SPM** doesn’t handle Xcode-specific flags, they are dropped. **`targetSettings`**, however, let **Tuist** inject these settings directly into the target’s build configuration, bypassing **SPM’s** limitations.

Additionally, **user-defined settings** like **`TESTING123`** are passed through because they are not tied to any specific toolchain or build system behavior, allowing **SPM** to treat them as arbitrary key-value pairs and pass them without interference.

### TLDR

SPM says this is handled by Xcode and not me and throws them away, it has no idea what `TESTING123` so it just passes it on

**Comment by kyleve** [comment]
Related, but it seems very surprising / scary to me that the _default_ behavior is to throw out warnings – I can see this biting people since they wouldn't expected.
